services/api-server/tests/unit/api_functions/test_api_routers_function_jobs.py

```diff
@@ -112,6 +113,50 @@ async def test_list_function_jobs(
     )
 
 
+async def test_list_function_jobs_with_job_id_filter(
+    client: AsyncClient,
+    mock_handler_in_functions_rpc_interface: Callable[[str, Any], MockType],
+    mock_registered_project_function_job: RegisteredProjectFunctionJob,
+    user_id: UserID,
+    product_name: ProductName,
+    auth: httpx.BasicAuth,
+) -> None:
+
+    mocked_list_function_jobs = mock_handler_in_functions_rpc_interface(
+        "list_function_jobs",
+        (
+            [mock_registered_project_function_job for _ in range(3)],
+            PageMetaInfoLimitOffset(total=5, count=3, limit=3, offset=1),
+        ),
+    )
+    response = await client.get(
+        f"{API_VTAG}/function_jobs",
+        params={
+            "function_job_ids": [str(mock_registered_project_function_job.uid)],
+            "limit": 3,
+            "offset": 1,
+        },
+        auth=auth,
+    )
+    mocked_list_function_jobs.assert_called_once_with(
+        ANY,  # Dummy rpc client
+        filter_by_function_job_ids=[mock_registered_project_function_job.uid],
+        filter_by_function_job_collection_id=None,
+        filter_by_function_id=None,
+        pagination_offset=1,
+        pagination_limit=3,
+        product_name=product_name,
+        user_id=user_id,
+    )
+    assert response.status_code == status.HTTP_200_OK
+    data = response.json()["items"]
+    assert len(data) == 3
+    assert (
+        RegisteredProjectFunctionJob.model_validate(data[0])
+        == mock_registered_project_function_job
+    )
```

Comment:
I would suggest a test where you can go through the pagination like you know that there are 13 items in the list and you expect 4 groups of 3 and one final item in the last request for a total of 4 requests